### PR TITLE
CN-2796: KYC Failure reasons

### PR DESCRIPTION
**Pull request by nemanjakrstonic**

### CODE REVIEW

File: app/components/atoms/EmptyState/EmptyState.js [hunk 1]
```diff
@@ -76,8 +80,10 @@ class EmptyState extends Component {
           <Image source={image || require("../../../../assets/images/diane-sad.png")}
             style={{ width: 140, height: 140, resizeMode: "contain" }} />
         </View>
-
-        <CelText margin="10 0 0 0" align="center" type="H3" weight={"500"} color={kycColor} bold>{title && title(kyc) || ''}</CelText>
+        <TouchableOpacity onPress={() => actions.openModal(MODALS.KYC_REJECTED_MODAL)} style={{ flex: 1, flexDirection: 'row', alignContent: 'center', alignItems: 'center' }} >
```
Comment (fjovakaric) [inline app/components/atoms/EmptyState/EmptyState.js]: Should be CelButton basic or transparent

File: app/components/organisms/KycRejectedModal/KycRejectedModal.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// import ContactSupport from '../../atoms/ContactSupport/ContactSupport';
+
+@connect(
+  state => ({
+    kycReasons: state.user.profile.kyc.rejectionReasons,
+
+  }),
+  dispatch => ({ actions: bindActionCreators(appActions, dispatch) }),
+)
+class KycRejectedModal extends Component {
```
Comment (fjovakaric) [inline app/components/organisms/KycRejectedModal/KycRejectedModal.js]: Reuse InfoModal component

File: app/components/atoms/EmptyState/EmptyState.js [hunk 3]
```diff
@@ -43,62 +44,134 @@ class EmptyState extends Component {
     this.state = emptyStateUtil(props.purpose, props.actions);
   }
 
-  // timer = () => {
-  //   this.setState({timer: null, counter: 0})
-  // }
-
+  renderErrors = () =>
+    this.props.kycReasons.map((reason, e) => (
```
Comment (StefanBurscher) [inline app/components/atoms/EmptyState/EmptyState.js]: change this e param to a proper name

File: app/components/atoms/EmptyState/EmptyState.js [hunk 4]
```diff
@@ -43,62 +44,134 @@ class EmptyState extends Component {
...
-    if (kycStatus === 'rejected' || kycStatus === 'rejeceted') {
-      kyc = 'Rejected'
-      kycColor = STYLES.COLORS.RED
+    const { title } = emptyStateProps;
+    const { kycStatus, actions } = this.props;
+    let kycColor = STYLES.COLORS.CELSIUS_BLUE;
+    let kyc = "";
+
+    if (kycStatus === "rejected" || kycStatus === "rejeceted") {
+      kyc = "Identity verification failed    ";
```
Comment (StefanBurscher) [inline app/components/atoms/EmptyState/EmptyState.js]: Remove unnecessary spacing

File: app/components/atoms/EmptyState/EmptyState.js [hunk 5]
```diff
@@ -43,62 +44,134 @@ class EmptyState extends Component {
...
+      </CelButton>
+    ) : (
+      <CelText
+        margin="0 0 0 0"
+        align="center"
+        type="H3"
+        weight={"500"}
+        color={kycColor}
+        bold
+      >
```
Comment (StefanBurscher) [inline app/components/atoms/EmptyState/EmptyState.js]: remove not existing props (bold)

File: app/components/atoms/EmptyState/EmptyState.js [hunk 6]
```diff
@@ -43,62 +44,134 @@ class EmptyState extends Component {
...
-
+  renderErrors = () =>
+    this.props.kycReasons.map((reason, e) => (
+      <CelText key={e} margin={"0 0 10 0"}>
+        {reason}
+      </CelText>
+    ));
 
-  render() {
+  renderKYCReject = () => {
```
Comment (StefanBurscher) [inline app/components/atoms/EmptyState/EmptyState.js]: renderKYCStatus

File: app/components/atoms/EmptyState/EmptyState.js [hunk 7]
```diff
@@ -43,62 +44,134 @@ class EmptyState extends Component {
...
+      image,
+      heading,
+      paragraphs,
+      onPress,
+      button,
+      support
+    } = emptyStateProps;
+    const { kycStatus, kycReasons, actions } = this.props;
+    const style = EmptyStateStyle();
+    const RenderKYCReject = this.renderKYCReject;
```
Comment (StefanBurscher) [inline app/components/atoms/EmptyState/EmptyState.js]: const KYCStatus = this.renderKYCStatus

File: app/components/molecules/InfoModal/InfoModal.js [hunk 8]
```diff
@@ -1,40 +1,71 @@
...
 
-      { noCopy && onNo ? (
-        <CelButton margin="10 0 10 0" onPress={onNo} basic>{noCopy}</CelButton>
+      {noCopy && onNo ? (
+        <CelButton margin="10 0 10 0" onPress={onNo} basic>
+          {noCopy}
+        </CelButton>
       ) : null}
+
+      {support ? <ContactSupport align="left" marginLeft={20} /> : null}
```
Comment (StefanBurscher) [inline app/components/molecules/InfoModal/InfoModal.js]: marginLeft is a nonexisting prop

File: app/components/screens/KYCLanding/KYCLanding.js [hunk 9]
```diff
@@ -32,184 +34,230 @@ const progressSteps = ['Verify Identity']
...
-    actions.getKYCStatus()
-    actions.getAllTransactions({ type: 'celpay' })
-    actions.getWalletSummary()
+    actions.getKYCStatus();
+    actions.getAllTransactions({ type: "celpay" });
+    actions.getWalletSummary();
   }
 
+  renderErrors = () =>
+    this.props.kycReasons.map((reason, e) => (
```
Comment (StefanBurscher) [inline app/components/screens/KYCLanding/KYCLanding.js]: e to proper prop name

File: app/components/screens/KYCLanding/KYCLanding.js [hunk 10]
```diff
@@ -32,184 +34,230 @@ const progressSteps = ['Verify Identity']
...
           <View style={style.stepsWrapper}>
-            {kycErrors && kycErrors.map(e => (
-              <CelText align={'center'} type={'H4'} weight={'300'} key={e}>
-                {e}
-              </CelText>
-            ))}
-            <View style={{ marginHorizontal: '6%'}}>
-            {this.renderProgressSteps(kycStatus)}
+            {kycErrors &&
+              kycErrors.map(e => (
```
Comment (StefanBurscher) [inline app/components/screens/KYCLanding/KYCLanding.js]: rename e to proper param name

File: app/components/screens/KYCLanding/KYCLanding.js [hunk 11]
```diff
@@ -32,184 +34,230 @@ const progressSteps = ['Verify Identity']
...
             </View>
           </View>
         </View>
       </View>
-    )
-  }
+    );
+  };
 
   renderStatus = () => {
```
Comment (StefanBurscher) [inline app/components/screens/KYCLanding/KYCLanding.js]: Don't use two functions with the same logic

File: app/components/screens/KYCLanding/KYCLanding.js [hunk 12]
```diff
@@ -1,25 +1,27 @@
...
+import CelText from "../../atoms/CelText/CelText";
+import RegularLayout from "../../layouts/RegularLayout/RegularLayout";
+import { THEMES, MODALS } from "../../../constants/UI";
+import STYLES from "../../../constants/STYLES";
+import { KYC_STATUSES } from "../../../constants/DATA";
+import Card from "../../atoms/Card/Card";
+import CelButton from "../../atoms/CelButton/CelButton";
+import InfoModal from "../../molecules/InfoModal/InfoModal";
+
+const progressSteps = ["Verify Identity"];
```
Comment (StefanBurscher) [inline app/components/screens/KYCLanding/KYCLanding.js]: Change this,  array is not needed

File: app/components/screens/KYCLanding/KYCLanding.js [hunk 13]
```diff
@@ -32,184 +34,230 @@ const progressSteps = ['Verify Identity']
...
         {this.renderKycStatus(kycStatus)}
+        {kycReasons ? (
+          <InfoModal
+            name={MODALS.KYC_REJECTED_MODAL}
+            heading="Identity verification failed"
+            support
+            yesCopy="Verify identity again"
+            onYes={actions.closeModal}
+          >
+            <>{this.renderErrors()}</>
```
Comment (StefanBurscher) [inline app/components/screens/KYCLanding/KYCLanding.js]: Change this to a const component

File: app/components/screens/KYCLanding/KYCLanding.js [hunk 14]
```diff
@@ -32,184 +32,200 @@ const progressSteps = ['Verify Identity']
...
+    <CelText key={index} margin={"0 0 10 0"}>
+      {reason}
+    </CelText>
+  ));
+ 
   render() {
-    const { kycStatus } = this.props
-    if (kycStatus === KYC_STATUSES.passed) return null
+    const { kycStatus, kycReasons, actions } = this.props;
+    // const Errors = this.renderErrors()
```
Comment (StefanBurscher) [inline app/components/screens/KYCLanding/KYCLanding.js]: why this comment ?

File: app/components/screens/KYCLanding/KYCLanding.js [hunk 15]
```diff
@@ -32,184 +32,200 @@ const progressSteps = ['Verify Identity']
...
-          {this.renderStatus()}
+        <View
+          style={{
+            flex: 1,
+            alignContent: "center",
+            justifyContet: "center",
+            marginBottom: 10
+          }}
+        >
+          {kycStatus === "rejected" || kycStatus === "rejeceted" ? (
```
Comment (StefanBurscher) [inline app/components/screens/KYCLanding/KYCLanding.js]: replace with kycStatus === KYC_STATUSES.rejected || kycStatus === KYC_STATUSES.rejeceted

File: app/components/atoms/EmptyState/EmptyState.js [hunk 16]
```diff
@@ -43,63 +44,131 @@ class EmptyState extends Component {
...
+            >
+              {paragraph}
+            </CelText>
+          ))}
 
         {button && onPress ? (
-          <CelButton margin="10 0 10 0" disabled={["pending", "sending", "sent"].indexOf(kycStatus) !== -1} onPress={onPress}>{button}</CelButton>
+          <CelButton
+            margin="10 0 10 0"
+            disabled={["pending", "sending", "sent"].indexOf(kycStatus) !== -1}
```
Comment (StefanBurscher) [inline app/components/atoms/EmptyState/EmptyState.js]: use KYC_STATUSES.pending ...

**Pull request merged**